util/threadpool_imp.cc

```diff
@@ -110,6 +134,13 @@ struct ThreadPoolImpl::Impl {
 
  int total_threads_limit_;
  std::atomic_uint queue_len_;  // Queue length. Used for stats reporting
+ // Number of reserved threads, managed by ReserveThreads(..) and
+ // ReleaseThreads(..), if a thread id is beyond total_threads_limit_ -
+ // reserved_threads_, its thread will be blocked to ensure the reservation
+ // mechanism
+ int reserved_threads_;
+ // Number of waiting threads (Maximum number of threads that can be reserved)
```

Comment:
I suggest adding the " in rare cases, num_waiting_threads_ could be less than reserved_threads
due to SetBackgroundThreadInternal or last excessive threads." comment here instead so future developer can be aware of this just by looking at these two variables declaration.

Reply:
Yeah, will add it.